### PR TITLE
Split into subprojects for compiler and language server

### PR DESCRIPTION
**Pull request by bentsherman**
The build seems to work:
```bash
./gradlew build
```

And produces the following files:
```console
$ find build/
build/
build/libs
build/libs/language-server.jar
build/tmp
build/tmp/jar
build/tmp/jar/MANIFEST.MF
```

But the manifest seems to be wrong:
```console
$ java -jar build/libs/language-server.jar 
no main manifest attribute, in build/libs/language-server.jar
```

I think something is wrong with the shadow JAR build, even though I didn't change any settings for it in the gradle build

### CODE REVIEW

**Comment by bentsherman** [comment]
Also, it used to produce a `language-server-all.jar` but now it doesn't

File: modules/compiler/build.gradle [hunk 1]
```diff
@@ -0,0 +1,18 @@
+plugins {
+  id 'antlr'
+}
+
+dependencies {
+  antlr 'org.antlr:antlr4:4.9.2'
+  implementation 'com.google.guava:guava:33.3.1-jre'
+  api 'org.apache.groovy:groovy:4.0.24'
+  api 'org.apache.groovy:groovy-json:4.0.24'
```
Comment (bentsherman) [inline modules/compiler/build.gradle:9]: TODO: move some code from language-server into compiler and change these back to `implementation`

File: build.gradle [hunk 2]
```diff
@@ -1,39 +1,18 @@
...
-}
-
-repositories {
-  mavenCentral()
-}
+allprojects {
+  java {
+    sourceCompatibility = JavaVersion.VERSION_17
+    targetCompatibility = JavaVersion.VERSION_17
+  }
```
Comment (pditommaso) [inline build.gradle:9]: This is a bit old style, now gradle allows sharing build logic using "convention" plugins. See here https://docs.gradle.org/current/samples/sample_convention_plugins.html#organizing_build_logic

see an example [here](https://github.com/seqeralabs/libseqera/blob/403ab8ac8813e79b6815df3b032feed0928a24fa/buildSrc/src/main/groovy/io.seqera.groovy-application-conventions.gradle#L14-L14) and usage [here](https://github.com/seqeralabs/libseqera/blob/403ab8ac8813e79b6815df3b032feed0928a24fa/lib-mail/build.gradle#L2-L3)

Comment (bentsherman) [reply build.gradle]: done

**Comment by bentsherman** [comment]
I added a convention plugin and a task to copy the shadow JAR to the top-level

Now I am testing the gradle build for the vs code extension and getting this error:
```
Could not determine the dependencies of task ':language-server:copyJar'.
> Task with path ':modules:language-server:build' not found in project ':language-server'.
```

The vs code extension has this `settings.gradle`:
```groovy
rootProject.name = 'vscode-language-nextflow'
include 'language-server'
```

And this gradle task to copy the shadow JAR into the extension build:
```groovy
task copyJar(type: Copy) {
  dependsOn ':language-server:build'
  from "$projectDir/language-server/build/libs/language-server-all.jar"
  into "$buildDir/bin"
}
```

For some reason, gradle can't find the `build` task for the language server subprojects when building from the vs code extension

**Comment by bentsherman** [comment]
I need to keep moving with other things so I'm going to merge what I have for now. I can make it work by manually building the language server JAR and only copying it from the vs code build:

```groovy
// settings.gradle (vs code extension)
rootProject.name = 'vscode-language-nextflow'
// include 'language-server'

// build.gradle (vs code extenstion)
task copyJar(type: Copy) {
  // dependsOn ':language-server:build'
  from "$projectDir/language-server/build/libs/language-server-all.jar"
  into "$buildDir/bin"
}
```

```bash
# from the vs code repo
cd language-server
./gradlew build
cd ..
./gradlew build
```

I suspect there is just something wrong in the language server gradle build. If either of you have any more pointers or can spot the issue, please let me know

**Pull request merged**